controllers/snapshot/snapshot_adapter.go

```diff
@@ -413,7 +413,13 @@ func (a *Adapter) EnsureAllReleasesExist() (controller.OperationResult, error) {
 		gitops.SetSnapshotIntegrationStatusAsError(a.snapshot, "Failed to get all ReleasePlans: "+err.Error())
 		a.logger.LogAuditEvent("Snapshot integration status marked as Invalid. Failed to get all ReleasePlans",
 			a.snapshot, h.LogActionUpdate)
-		return controller.RequeueOnErrorOrStop(a.client.Status().Patch(a.context, a.snapshot, patch))
+		er := a.client.Status().Patch(a.context, a.snapshot, patch)
+		if er != nil {
+			a.logger.Error(er, "Failed to mark snapshot integration status as invalid",
+				"snapshot.Name", a.snapshot.Name)
+			return controller.RequeueWithError(er)
```

Comment:
This hides why we even get to this error, we should expose both errors, we can use Join function https://pkg.go.dev/errors#Join

Reply:
Expose? Do you mean expose `er` and `err` here together? The `err` is logged out above and will retry `controller.RequeueWithError(err)` below

Reply:
Yes, exposing errors together